lib/puppet/parser/functions/clamps_user_number.rb

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env ruby
+
+module Puppet::Parser::Functions
+  newfunction(:clamps_user_number, :type => :rvalue) do |args|
+    username = args.first.to_s
+
+    number = username.sub('user', '')
```

Comment:
Might be worth documenting via comments that usernames are expected to be `=~ /^user\d+$/`, or providing a default of `rand(60)` if the username doesn't meet this expectation.